qualtran/bloqs/arithmetic/permutation_test.py

```diff
@@ -44,16 +48,38 @@
 from qualtran.symbolics import ceil, log2, slen
 
 
-def test_examples(bloq_autotester):
-    bloq_autotester(_permutation_cycle)
-    bloq_autotester(_permutation)
-    bloq_autotester(_sparse_permutation)
-
-
-def test_symbolic_examples(bloq_autotester):
-    bloq_autotester(_permutation_cycle_symb)
-    bloq_autotester(_permutation_symb)
-    bloq_autotester(_permutation_symb_with_cycles)
+@pytest.mark.parametrize(
+    "bloq_ex",
+    [
+        _permutation_cycle,
+        _permutation,
+        _sparse_permutation,
+        _permutation_cycle_symb,
+        _permutation_cycle_symb_N,
+        _permutation_symb,
+        _permutation_symb_with_cycles,
+        _sparse_permutation_with_symbolic_N,
+    ],
+    ids=lambda bloq_ex: bloq_ex.name,
+)
+def test_examples(bloq_autotester, bloq_ex):
+    bloq_autotester(bloq_ex)
+
+
+@pytest.mark.parametrize(
+    "bloq_ex",
+    [
+        _permutation_cycle,
+        _permutation,
+        _sparse_permutation,
+        _permutation_cycle_symb_N,
+        _permutation_symb_with_cycles,
+        _sparse_permutation_with_symbolic_N,
+    ],
+    ids=lambda bloq_ex: bloq_ex.name,
+)
+def test_decomposition(bloq_ex):
+    qlt_testing.assert_valid_bloq_decomposition(bloq_ex.make())
```

Comment:
this is redundant with the `bloq_autotester`

Reply:
I want to verify that the bloq actually decomposes (i.e. without raising a `DecomposeTypeError`). `bloq_autotester` treats `DecomposeTypeError` as a success right?

Reply:
oh good point. (technically it would report it as a skip)